gossip/privdata/pvtdataprovider.go

```diff
@@ -202,32 +202,48 @@ func (pdp *PvtdataProvider) RetrievePvtdata(pvtdataToRetrieve []*ledger.TxPvtdat
 
 	pvtdata := make(rwsetByKeys)
 
+	//If there is no private data to retrieve for the block, skip all population attempts and return
+	if len(pvtdataRetrievalInfo.remainingEligibleMissingKeys) == 0 {
+		pdp.logger.Debugf("No eligible collection private write sets to fetch for block [%d]", pdp.blockNum)
+		retrievedPvtdata.pvtdataRetrievalInfo = pvtdataRetrievalInfo
+		retrievedPvtdata.blockPvtdata = pdp.prepareBlockPvtdata(pvtdata, pvtdataRetrievalInfo)
+		return retrievedPvtdata, nil
+	}
+
+	fetchStats := fetchStats{}
```

Comment:
`fetchStats := &fetchStats{}` would be more appropriate and idiomatic use

Reply:
done.